swarm/storage/mru/resource.go

```diff
@@ -1059,21 +991,16 @@ func isMultihash(data []byte) int {
 	return cursor + inthashlength
 }
 
-func NewTestHandler(datadir string, params *HandlerParams) (*Handler, error) {
-	path := filepath.Join(datadir, DbDirName)
-	rh, err := NewHandler(params)
-	if err != nil {
-		return nil, fmt.Errorf("resource handler create fail: %v", err)
-	}
-	localstoreparams := storage.NewDefaultLocalStoreParams()
-	localstoreparams.Init(path)
-	localStore, err := storage.NewLocalStore(localstoreparams, nil)
-	if err != nil {
-		return nil, fmt.Errorf("localstore create fail, path %s: %v", path, err)
-	}
-	localStore.Validators = append(localStore.Validators, storage.NewContentAddressValidator(storage.MakeHashFunc(resourceHash)))
-	localStore.Validators = append(localStore.Validators, rh)
-	netStore := storage.NewNetStore(localStore, nil)
-	rh.SetStore(netStore)
-	return rh, nil
+// NewResourceHash will create a deterministic address from the update metadata
+// format is: hash(period|version|publickey|namehash)
+func NewResourceHash(period uint32, version uint32, namehash common.Hash, publicKeyBytes [publicKeyLength]byte) []byte {
```

Comment:
ah, i see this function is a misnomer. Call it `NewUpdateAddressPreimage` or something ;)

Reply:
Why is "resourcehash" a misnomer?